src/creds.rs

```diff
@@ -157,6 +103,14 @@ impl Credential {
             MLSCredentialType::X509(_) => panic!("X509 certificates are not yet implemented."),
         }
     }
+
+    /// Get the ciphersuite associated with the credential.
+    pub fn ciphersuite(&self) -> &Ciphersuite {
+        match &self.credential {
+            MLSCredentialType::Basic(basic_credential) => &basic_credential.ciphersuite,
```

Comment:
yeah... enums...

Reply:
Right. Credential should be a `trait`, I noticed that while writing this code, but I think we should leave that to a different PR. Should I file an issue?

Reply:
I think you already did a lot of the work when I asked for help w.r.t. `Sized` traits. Do you still have that code somewhere?

Reply:
Yeah this should be a different PR & issue. I have some code but have to look at it again.

Reply:
Filed issue #163 and labelled it P3 for now. Feel free to relabel.